ethereum/nft/trades/insert_nftx.sql

```diff
@@ -0,0 +1,290 @@
+Redeemed_t1 AS
+(
+    select * from nftx_v2."NFTXVaultUpgradeable_v1_evt_Redeemed"
+    UNION
+    select * from nftx_v2."NFTXVaultUpgradeable_v1_evt_Redeemed0"
+),
+Redeemed_t2 AS
+(
+    select
+    evt_block_time as time,
+    evt_block_number,
+    evt_tx_hash as tx,
+    evt_index,
+    'Redeem' as type,
+    'Buy' as tx_type,
+    contract_address as vault,
+    unnest("nftIds") as nfts,
+    contract_address as seller,
+    "to" as buyer
+    from
+    Redeemed_t1
+    ORDER BY time
+),
+Swapped_t1 AS
+(
+    select * from nftx_v2."NFTXVaultUpgradeable_v1_evt_Swapped"
+    UNION
+    select * from nftx_v2."NFTXVaultUpgradeable_v1_evt_Swapped0"
+),
+Swapped_t2M AS
+(
+    select
+    evt_block_time as time,
+    evt_block_number,
+    evt_tx_hash as tx,
+    evt_index,
+    'Swap' as type,
+    'Sell' as tx_type,
+    contract_address as vault,
+    unnest("nftIds") as nfts,
+    "to" as seller,
+    contract_address as buyer
+    from
+    Swapped_t1
+    ORDER BY time
+),
+Swapped_t2R AS
+(
+    select
+    evt_block_time as time,
+    evt_block_number,
+    evt_tx_hash as tx,
+    evt_index,
+    'Swap' as type,
+    'Buy' as tx_type,
+    contract_address as vault,
+    unnest("redeemedIds") as nfts,
+    contract_address as seller,
+    "to" as buyer
+    from
+    Swapped_t1
+    ORDER BY time
+),
+Txs_t as
+(
+    select * from Minted_t2
+    UNION ALL
+    select * from Redeemed_t2
+    UNION ALL
+    select * from Swapped_t2M
+    UNION ALL
+    select * from Swapped_t2R
+),
+Txs_t2 as
+(
+    select * from Txs_t
+    where time >= start_ts AND time < end_ts
+),
+Txs as
+(
+    select * from Txs_t2 LEFT JOIN vaults on Txs_t2.vault = vaults."vaultAddress"
+)
+
+-- #################################################################
+-- #################################################################
+
+
+-- #################################################################
+-- Parse Data
+-- #################################################################
+rows AS (
+    INSERT INTO nft.trades (
+        block_time,
+        nft_project_name,
+        nft_token_id,
+        erc_standard,
+        platform,
+        platform_version,
+        trade_type,
+        number_of_items,
+        category,
+        evt_type,
+        usd_amount,
+        seller,
+        buyer,
+        original_amount,
+        original_amount_raw,
+        original_currency,
+        original_currency_contract,
+        currency_contract,
+        nft_contract_address,
+        exchange_contract_address,
+        tx_hash,
+        block_number,
+        nft_token_ids_array,
+        senders_array,
+        recipients_array,
+        erc_types_array,
+        nft_contract_addresses_array,
+        erc_values_array,
+        tx_from,
+        tx_to,
+        trace_address,
+        evt_index,
+        trade_id
+    )
+
+    SELECT
+        time AS block_time,
+        name AS nft_project_name,
+        nfts AS nft_token_id,
+        'erc20' AS erc_standard,
```

Comment:
In NFTX vaults a NFT is converted to a erc20 token, user can take that erc20 token and use it to trade in parts or exchange, a user can return the said vaultToken(erc20) anytime to NFTX vault and redeem the NFT to cash out.
Since the whole platform operates using the ERC20 vault tokens, thats why i hardcoded it.